### PR TITLE
Pass certain fields as objects to the app, and use double-quotes in exam...

### PR DESCRIPTION
**Pull request by islemaster**
...ples to make them valid JSON

### CODE REVIEW

File: dashboard/app/helpers/levels_helper.rb [hunk 1]
```diff
@@ -256,7 +256,10 @@ def blockly_options(local_assigns={})
     level['scale'] = {'stepSpeed' =>  @level.properties['step_speed'].to_i } if @level.properties['step_speed'].present?
 
     # Blockly requires these fields to be objects not strings
-    %w(map initialDirt finalDirt goal soft_buttons inputOutputTable).each do |x|
+    (
+      %w(map initialDirt finalDirt goal soft_buttons inputOutputTable)
+      .concat NetSim.json_object_attrs
+    ).each do |x|
```
Comment (Bjvanminnen) [inline dashboard/app/helpers/levels_helper.rb]: should this be a method on the model maybe? (i.e. inputOutputTable belongs in calc.object_fields)

File: dashboard/app/models/net_sim.rb [hunk 2]
```diff
@@ -21,7 +32,7 @@ def self.skins
 
   # DNS modes, used by levelbuilder
   def self.dns_modes
-    ['none', 'manual', 'automatic']
+    %w( none manual automatic )
```
Comment (islemaster) [inline dashboard/app/models/net_sim.rb]: Just learned about this notation.  Neato!

File: dashboard/app/helpers/levels_helper.rb (hunk 1 above)
Comment (Bjvanminnen) [inline dashboard/app/helpers/levels_helper.rb]: I guess that's kind of what you did with NetSim. wondering if we should generalize.

Comment (islemaster) [inline dashboard/app/helpers/levels_helper.rb]: I was going to, then after investigating a little I realized that initialDirt and finalDirt are sort of hybrid/owned by both `Grid` and `Karel`, and I'm probably not the guy to untangle that.

**Pull request merged**